### DISCUSSION TITLE
Avoid re-indexing the same files?

### DISCUSSION THREAD

**Discussion by nitobuendia** [photoprism/photoprism#3451] [Q&A] [answered]
## Context
I have a Photoprism self-hosted instance running on docker. I manage my photos as read only using the `/originals/` folder. In order to make sure that new photos are indexed, I run the following command on a cron job: `docker exec photoprism /opt/photoprism/bin/photoprism index`. I am not passing any flags to indicate full indexing. My goal is to replicate the behaviour 

## Issues
In general, this works fine and skips the processed files and looks for new photos to index. My problem is that even when there are no new photos, the system keeps indexing or processing old duplicated files.

For example: `15:12:37 time="2023-06-04T07:12:37Z" level=info msg="index: skipped duplicate main jpg file 'albums/2022 - Spain/0471_IMG_20220503_142439.jpg'"`. This image was already processed before in the previous indexing. This is a problem because indexing takes a long time despite expecting it to take a few seconds since no new photos are found since I have a large number of duplicated photos (see my folder structure on #3450). However, I would have expected Photoprism to "store" that it has already processed the duplicate and not to re-process again.

This behaviour does not seem to be the case (at least not visible) when doing Library > Start (Rescan), which is my intended beahviour.

## Questions
Is there a way to avoid this problem?

Thanks in advance!

## Version Details

Photoprism version: `photoprism version 230603-378d4746a-Linux-AMD64-Plus` - installed via docker compose on an Ubuntu server machine.

**Comment by lastzero** [comment, marked as answer]
When a file is skipped, it is not indexed. I wonder how this can cause significant overhead? Is it possible that the mod timestamp of files has changed? This can lead to reindexing of existing files. If this happens, try `PHOTOPRISM_DISABLE_CHOWN: "true"` as this will prevent updates on startup, see https://docs.photoprism.app/getting-started/config-options/#docker-image.

**Comment by nitobuendia** [comment]
```
Is it possible that the mod timestamp of files has changed?
```

I don't think so. It doesn't re-index any photo that was indexed before. It only runs through the "duplicated" photos/folders.

Nonetheless, I can test. The job has been running for 4h since there is a large number of (duplicated) photos. Once it completes, I can test two things: (1) run it again immediately (without having added any pics or any changes to the server that could trigger a change in the mod timestamp); and (2) if that doesn't work, I will add the `disable_chown` setting to test if that has an impact.

**Comment by nitobuendia** [comment]
@lastzero Thanks. `PHOTOPRISM_DISABLE_CHOWN: "true"` has helped, but not fully. It's both converting files (e.g. heic to jpg [1]) as well as re-indexing duplicated files [2]. Problem [2], which is the one I was first reporting, is fixed. Problem [1] still remains.


I just run it again (before making any changes). There was no backups in between, and no changes of any form to the photos folder or to the server. This are the sample messages:

```
[1] 18:40:12 time="2023-06-04T10:40:12Z" level=info msg="convert: converting img_2417.heic to img_2417.heic.jpg (heif-convert)"
[2] 18:47:52 time="2023-06-04T10:47:52Z" level=info msg="index: skipped duplicate main jpg file 'albums/Shared - Mitsuko:Nito/5177_IMG_20200524_095159.jpg'"
```


Then, I've added `PHOTOPRISM_DISABLE_CHOWN: "true"` to the docker-compose.yaml file and restarted the container (`docker-compose down && docker-compose up -d`) and re-run the index command:

```
[1] 18:56:27 time="2023-06-04T10:56:27Z" level=info msg="convert: converting img_2417.heic to img_2417.heic.jpg (heif-convert)"
```


[2] This is fixed as it goes almost immediately to `19:07:22 time="2023-06-04T11:07:22Z" level=info msg="indexed 99,754 files in 11m12.252514111s"`.


Is there a way to fix [1] too?

Thanks in advance!

**Reply by lastzero** [reply]
From all I know 2 is only a log message that doesn't trigger any index changes. Also, I've never seen or heard about any performance issues with it. If you believe this is wrong, we can take a look when there is a bit of time. As the code is public, you could also see for yourself if there is a way to make it faster.

**Comment by nitobuendia** [comment]
Got it, thanks. The way it sounds is like it's converting the images from heic to jpg on every run, which is odd. In any case, this is now down from 5h30 to 10min (after disabling chown), so all good on my end. Thanks for your time.
